### PR TITLE
fix(providers): allow fields to be marked as automated

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Changes

Fixes https://linear.app/nango/issue/NAN-2280/ability-to-set-connectionconfig-as-automated

- Allow fields to be marked as automated
- Set salesforce instance url as automated

## Tests

I don't have a Salesforce account, if you have one I would appreciate a local test please (@khaliqgant or @hassan254-prog)

### CODE REVIEW

File: packages/shared/providers.yaml [hunk 1]
```diff
@@ -5169,7 +5169,7 @@ salesforce:
             description: The instance URL of your Salesforce account
             format: uri
             pattern: '^https?://.*$'
-            optional: true
+            automated: true
```
Comment (khaliqgant) [inline packages/shared/providers.yaml:5190]: Should update https://github.com/NangoHQ/nango/blob/master/packages/shared/providers.yaml#L2643 and github-app-oauth as well

Comment (bodinsamuel) [reply packages/shared/providers.yaml:5190]: both appLink and installationId or just installationId?

Comment (khaliqgant) [reply packages/shared/providers.yaml:5190]: Just installationId -> https://nangohq.slack.com/archives/C07216H4BRT/p1732707585655439?thread_ts=1732703023.390049&cid=C07216H4BRT

Comment (bodinsamuel) [reply packages/shared/providers.yaml:5190]: ah yes 🤦🏻

**Review by khaliqgant — APPROVED** [review]

**Comment by hassan254-prog** [comment]
> ## Changes
> Fixes https://linear.app/nango/issue/NAN-2280/ability-to-set-connectionconfig-as-automated
> 
> * Allow fields to be marked as automated
> * Set salesforce instance url as automated
> 
> ## Tests
> I don't have a Salesforce account, if you have one I would appreciate a local test please (@khaliqgant or @hassan254-prog)

@bodinsamuel, checking 👍.

**Comment by hassan254-prog** [comment]
@bodinsamuel , works perfect with salesforce 👍

**Pull request merged**